client/jetpack-onboarding/steps/homepage.jsx

```diff
@@ -21,6 +21,7 @@ import { saveJetpackOnboardingSettings } from 'state/jetpack-onboarding/actions'
 class JetpackOnboardingHomepageStep extends React.PureComponent {
 	handleHomepageSelection = homepageFormat => {
 		const { siteId } = this.props;
+		this.props.recordJpoEvent( 'calypso_jpo_' + homepageFormat + '_clicked' );
```

Comment:
Do you think it would be better if we can pass additional data to the event (as a second parameter of `recordJpoEvent`)?

This would allow us to have the event always named `'calypso_jpo_homepage_format_clicked'`, with an additional property of `homepageFormat: posts` or `homepageFormat: page`. For consistency reasons, I find it always better to have the same event name with additional properties for its different variations.

I think this will be useful for the rest of the steps, too.

I've created #21595 which allow support for additional properties to events.

Reply:
Right, I see your point. On the other hand, I reckon that here we are concerned more with easy Tracks segmentation than naming on the Calypso side. I'm not overly familiar with ins and outs of Tracks, but the recommendation is to not overload the event objects with too many properties.  How strongly do you feel about it being adjusted the way you mentioned? 😄

Reply:
Oh, my opinion is quite strong here 😉 

I don't think a single parameter contributes to overloading the event with too many parameters. I'll also support my opinion with with several additional reasons:

* Sustainability - if we decide to track something additional with this event, it will be much easier to add it to the object properties that already exist, rather than to the event name.
* Flexibility - your suggested approach will work because the `homepageFormat` variable is a string, but what happens if it is an integer, a boolean or a more complex one - (array/object)? It might not result in a very predictable event name after all.
* Searching for events - if we use static event names, it's easier to locate them throughout Calypso whenever we want to change how they work OR when we're building for funnels and looking for events.
* Consistency - that's how we're doing it pretty much everywhere throughout Calypso.

Reply:
This is quite a strong argument for the change 😆  Point taken and appreciated 👍